examples/bri-3/src/bri/workgroup/workgroups/agents/workgroups.agent.ts

```diff
@@ -0,0 +1,114 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BpiSubject } from 'src/bri/identity/bpiSubjects/models/bpiSubject';
+import { Workflow } from '../../workflows/models/workflow';
+import { Workstep } from '../../worksteps/models/workstep';
+
+import { uuid } from 'uuidv4';
+import { Workgroup } from '../models/workgroup';
+import { BpiSubjectStorageAgent } from '../../../identity/bpiSubjects/agents/bpiSubjectsStorage.agent';
+import { WorkgroupStorageAgent } from './workgroupStorage.agent';
+import {
+  BPISUBJECT_NOT_FOUND_ERR_MESSAGE,
+  WORKGROUP_NOT_FOUND_ERR_MESSAGE,
+} from '../api/err.messages';
+
+// Agent methods have extremely declarative names and perform a single task
+@Injectable()
+export class WorkgroupAgent {
+  constructor(
+    private workgroupStorageAgent: WorkgroupStorageAgent,
+    private bpiSubjectStorageAgent: BpiSubjectStorageAgent,
+  ) {}
+
+  public async fetchWorkgroupAdministratorsAndThrowIfNoneExist(
+    administrstorIds: string[],
+  ): Promise<BpiSubject[]> {
+    const bpiSubjects = await this.bpiSubjectStorageAgent.getBpiSubjectsById(
+      administrstorIds,
+    );
+
+    if (Array.isArray(bpiSubjects) && !bpiSubjects.length) {
```

Comment:
i think we should not silently skip in case one of the administrator id does not exits. Better would be to check if the length of the provided list is the same as length of the fetched bpiSubjects, and fail in case it is not.

Reply:
Array.isArray(bpiSubjects) is not needed, storage agent method will use map which will always return array

agree on comparing the lists length (even though nothing prevents from sending duplicates in `administrstorIds` making that check not precise enough, this can happen since ids are in dto so we dont control what is sent), maybe better is to compare if arrays are the same

Reply:
Checks length first, then checks id by id if a matching id exists.